### DISCUSSION TITLE
Dual Y motors

### DISCUSSION THREAD

**Discussion by HalbesHaehnchen** [phil-barrett/grblHAL-teensy-4.x#91] [Q&A] [answered]
Good day again,

is there any benefit when using "ganged" Y motors when using web builder? I am not looking into auto square feature as my mechanical build moves pretty squared. Want to keep it simple for now. 
At the moment I just wired the Y output of the board to two Stepper drivers and it seem to work fine at first test.

**Comment by phil-barrett** [comment]
Ignoring the issue of maintaining gantry squareness for a minute, the benefit of using separate step/dir/enable signals for each driver is that some stepper drivers, often older ones, draw more current than modern driver do. Doubling that (by connecting one output to two stepper drivers) can stress or even overwhelm the output circuits on the breakout board.

As to gantry squareness, machines wear and flex.  Even the most rigid ones have some flex.  You may not see out of square conditions immediately but it can happen over time.  In particular, if you have a moving gantry with 2 Y motors, and you place a large asymmetrical force on either end of the gantry, it is possible to lose steps on one of the Y motors.  Once you do that, it won't "snap" back into place.  Auto-squaring will fix it.  It's your choice but I highly recommend auto-squaring.

**Reply by HalbesHaehnchen** [reply]
Ah okay, understood that. Should have mentioned that even I am using a CNC milling frame, it will just run 2 Laser modules. So no additional force while the machine is working. Just its weight moving around. 
But when I should not connect 2 drivers to the same output I can go the auto square way anyway. Its just an additional limit switch and some software calibration later right? As far as I understood it will be A axis/limits for the second stepper then? No need for cutting solder pads on the backside?

**Comment by phil-barrett** [comment, marked as answer]
OK, so running a machine for laser cutting/etching means you won't have high stresses on the gantry.  I can see why you wouldn't want to bother with auto-squaring.

Yes, connect to each stepper driver independently. 

On your question about setting up auto-squaring. If you already have homing set up, to add auto-square is literally 1 extra limit sensor and associated wiring/connector.  The software part is simple - build a grblHAL version with auto-squaring.  When homing, it will do the auto-squaring as well. No board hackery need.  Don't know if you have seen [my auto-squaring blog](https://www.grbl.org/single-post/it-s-hip-to-be-auto-square).
